### PR TITLE
Issue #4650 - do not use ServiceLoader every time a WSSession is started

### LINKED ISSUE #4650: Do not use ServiceLoader every time a WebSocketSession is created
**Jetty version**
`9.4.27.v20200227`

**Java version**
```
openjdk version "13.0.2" 2020-01-14
OpenJDK Runtime Environment Zulu13.29+9-CA (build 13.0.2+6-MTS)
OpenJDK 64-Bit Server VM Zulu13.29+9-CA (build 13.0.2+6-MTS, mixed mode, sharing)
```

**OS type/version**
`Ubuntu 18.04.4 LTS`

**Description**

Hey, Jetty Maintainers! After running a benchmark with a few thousands of WebSocket clients, we noticed in the CPU profile (it was also very noticeable in the allocation profile) that jetty scans jars looking for `org.eclipse.jetty.websocket.common.RemoteEndpointFactory` for every new accepted connection. 

Screen Shot 2020-03-05 at 2 35 24 pm

The graph corresponds to the following lines in `WebSocketSession`. Overriding this behaviour doesn't seem to be easy since a concrete class is created in the session factory. 

https://github.com/eclipse/jetty.project/blob/67eb9b35311dfd67b149204105a4f980772fa755/jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/WebSocketSession.java#L171-L172

As far as I can tell from browsing the code, this is not an issue for the `10.x` branch. However, it would be great to resolve it in `9.4.x` somehow. As far as I can tell, there are a few options: 
* Perform the lookup statically once on the startup. 
* Provide a configuration option to disable this lookup. 

What do you think? If any of the above sounds good to you, I'll be happy to submit a PR.

### PR DESCRIPTION
**Pull request by lachlan-roberts**
Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

### CODE REVIEW

**Review by gregw — APPROVED** [review]
LGTM

File: jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/WebSocketSession.java [hunk 1]
```diff
@@ -83,6 +84,15 @@
     private UpgradeResponse upgradeResponse;
     private CompletableFuture<Session> openFuture;
 
+    static
+    {
+        // Attempt to discover a RemoteEndpointFactory with the SerivceLoader.
+        Iterator<RemoteEndpointFactory> iter = ServiceLoader.load(RemoteEndpointFactory.class).iterator();
```
Comment (joakime) [inline jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/WebSocketSession.java:90]: Is this running under the correct Classloader / Thread Context?

Comment (lachlan-roberts) [reply jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/WebSocketSession.java:90]: Yes, its running on the same thread and with the same contextClassLoader it has for `doStart()`.

**Review by joakime — APPROVED** [review]
I'm good with this.

**Pull request merged**